packages/mui-material/src/Grid/Grid.js

```diff
@@ -369,14 +369,12 @@ const Grid = React.forwardRef(function Grid(inProps, ref) {
   const columns = container ? columnsProp || 12 : columnsContext;
 
   const breakpointsValues = {};
-  const propsRestFiltered = { ...propsRest };
+  const otherFiltered = { ...other };
 
-  breakpoints.keys.forEach((key) => {
-    if (key in propsRest) {
```

Comment:
this is why the `key` prop was added TypeScript has some dark magic that makes him think that a key prop might be present here. If you remove `key in propsRest` then we are good. As a side note, we might have preferred using `hasOwnProperty` per https://masteringjs.io/tutorials/fundamentals/hasownproperty. In our case, I could get away with a nulish check.

Reply:
`hasOwnProperty` makes sense too, but the nullish check reads better to me 👍